cmake/winml_sdk_helpers.cmake

```diff
@@ -16,7 +16,22 @@ function(get_installed_sdk
     set(${sdk_folder} ${win10_sdk_root} PARENT_SCOPE)
 
     # return the sdk version
-    set(${output_sdk_version} ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION} PARENT_SCOPE)
+    if(CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION)
+        set(${output_sdk_version} ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION} PARENT_SCOPE)
+    else()
+        # choose the SDK matching the system version, or fallback to the latest
+        file(GLOB win10_sdks RELATIVE "${win10_sdk_root}/UnionMetadata" "${win10_sdk_root}/UnionMetadata/*.*.*.*")
+        list(GET win10_sdks 0 win10_preferred_sdk)
+        foreach(sdk IN LISTS win10_sdks)
+            string(FIND ${sdk} ${CMAKE_SYSTEM_VERSION} is_system_version)
+            if(NOT ${is_system_version} EQUAL -1)
+                set(${output_sdk_version} ${sdk} PARENT_SCOPE)
+                return()
+            elseif(sdk VERSION_GREATER win10_preferred_sdk)
+                set(${output_sdk_version} ${sdk} PARENT_SCOPE)
+            endif()
+        endforeach()
+    endif()
```

Comment:
Ninja doesn't have a target SDK config like VS, so if the user doesn't explicitly give one with winml_WINDOWS_SDK_DIR_OVERRIDE / winml_WINDOWS_SDK_VERSION_OVERRIDE I'm picking one by default. I chose to pick one matching the current system version, or fallback to the latest one.

If someone doesn't think this is a sane default and has better suggestions (or thinks we should just fail the build and ask the user for an explicit version) let me know.